gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/FSSpecStore.java

```diff
@@ -98,17 +98,39 @@ public FSSpecStore(Config sysConfig, SpecSerDe specSerDe, Optional<Logger> log)
     }
   }
 
+  public static String getFlowGroup(Path specUri){
+    return specUri.getParent().getName();
+  }
+
+  public static String getFlowName(Path specUri){
+    return Files.getNameWithoutExtension(specUri.getName());
+  }
+
   @Override
   public boolean exists(URI specUri) throws IOException {
     Preconditions.checkArgument(null != specUri, "Spec URI should not be null");
-
+    Path flowPath = new Path(specUri.getPath());
+    String flowGroup = getFlowGroup(flowPath);
+    String flowName = getFlowName(flowPath);
```

Comment:
We should be able to check for the existence of <store_root>/<flowGroup> first then list that directory to avoid listing from the store_root. You can directly call listStatus on <store_root>/<flowGroup> and catch the FileNotFoundException and return false.

Reply:
Will do.